File: src/__tests__/golden/fixtures/ts-layered/controllers/user-controller.ts
import { getUser, createUser } from "../services/user-service";
import { formatUser } from "../utils/format";
import { log } from "../utils/logger";

export function handleGetUser(id: string): string {
  log(`GET /users/${id}`);
  const user = getUser(id);
  return formatUser(user);
}

export function handleCreateUser(name: string, email: string): string {
  log(`POST /users`);
  const user = createUser(name, email);
  return formatUser(user);
}

File: src/__tests__/golden/fixtures/java-service/src/main/java/com/example/controller/UserController.java
package com.example.controller;

import com.example.model.User;
import com.example.service.UserService;

public class UserController {
    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    public User handleGetUser(Long id) {
        return userService.getUser(id);
    }

    public User handleCreateUser(String name, String email) {
        return userService.createUser(name, email);
    }
}
